builtin/providers/ultradns/provider.go

```diff
@@ -0,0 +1,50 @@
+package ultradns
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/Ensighten/udnssdk"
+)
+
+// Provider returns a terraform.ResourceProvider.
+func Provider() terraform.ResourceProvider {
+	return &schema.Provider{
+		Schema: map[string]*schema.Schema{
+			"username": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				DefaultFunc: schema.EnvDefaultFunc("ULTRADNS_USERNAME", nil),
+				Description: "UltraDNS Username.",
+			},
+
+			"password": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				DefaultFunc: schema.EnvDefaultFunc("ULTRADNS_PASSWORD", nil),
+				Description: "UltraDNS User Password",
+			},
+			"baseurl": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				DefaultFunc: schema.EnvDefaultFunc("ULTRADNS_BASEURL", udnssdk.DefaultTestBaseURL),
+				Description: "UltraDNS Base Url(defaults to testing)",
```

Comment:
Seems odd to default this to a test value. Perhaps better to just leave it required so that omission is an error?